includes/class-wc-gateway-stripe.php

```diff
@@ -294,6 +294,22 @@ public function payment_fields() {
 			$pay_button_text = '';
 		}
 
+		/**
+		 * Start a payment intent.
+		 *
+		 * @see https://stripe.com/docs/payments/payment-intents/quickstart
+		 */
+		$request = array(
+			'amount'               => round( $total * 100 ),
+			'capture_method'       => 'manual',
+			'currency'             => get_woocommerce_currency(),
+			'allowed_source_types' => array(
+				'card',
+			),
+		);
+		$intent        = WC_Stripe_API::request( $request, 'payment_intents' );
```

Comment:
*I'm just brain-dumping all this here, we don't need to go into this for now.*

I don't like blocking the rendering of the checkout page to wait for an external API call. Specially since Stripe may not even be the selected payment gateway.

Let's go ahead for now, but I want to revisit this before launch and see if there's any way around it. Maybe adding an AJAX endpoint to fetch the Intent and calling it when the `stripe.js` form is initializing, so at least it won't block the rest of the checkout page.

Also, Stripe recommends to only create 1 Intent per order. As we talked, the recommended flow is:
* User submits checkout with shipping/billing address and products.
* Order is created.
* Payment Intent is created.
* User is presented with Payment Fields.
* User submits the payment fields and does the whole 3DS dance.

Obviously this is a big departure from the current checkout, and probably the flow you've implemented is preferred, but I'm concerned about this part:
> A PaymentIntent tracks the process of collecting a payment from your customer. We recommend that you create exactly one PaymentIntent for each order or customer session in your system. **You can reference the PaymentIntent later to see the history of payment attempts for a particular session.**

As I can interpret it, that means that every single Intent will be visible in the merchan't Stripe dashboard. Which means that a merchant using Stripe in WooCommerce will potentially get thousands of Intents even if he just got a handful of actual orders.

PayPal does a "1-button checkout", so PayPal's flow is more straight-forward from our point of view:
* User fills all their (non-payment) data.
* User clicks the PayPal button.
* Everything else happens outside our control.

Putting the credit card fields in the same page as the rest of the checkout certainly complicates things. Maybe having something like this would be better?
* User fills all their non-payment data.
* Instead of the payment fields, in the "Pay with Stripe" section of the checkout there's a call-to-action like `PAY WITH STRIPE`.
* When clicking that button, an AJAX call is performed to create a Payment Intent.
* When that call finishes, the button is replaced with the Stripe payment fields, which are initialized at that moment.
* The rest of the flow wouldn't change.

Reply:
> As I can interpret it, that means that every single Intent will be visible in the merchan't Stripe dashboard.

Just checked and yes, every intent appears in the dashboard as `Incomplete`, which might scare a lot of merchants.

The way to avoid this is much simpler though: The intent `client_secret` is only needed when calling `handleCardPayment`, which only happens once someone clicks the _Place order_ button.

As we mentioned in Slack, it will be much easier and more practical to have an additional AJAX call to the local server in order to create the intent directly before calling `handleCardPayment`.

I have added a `ToDo` comment for this.